### PR TITLE
Adds flatten effect to clown car running over people

### PR DESCRIPTION
**Pull request by TerraGS**
https://user-images.githubusercontent.com/40574831/191116005-5fbd8bbe-61fe-4af4-a60a-a5e67743cbfa.mp4

## About The Pull Request
Adds a new mob interaction to the clown car that cartoonishly flattens anyone it rolls over and leaves them momentarily immobilized. Ultimately harmless like any good prank.

## Why It's Good For The Game
More wacky clown antics add to the fun of the round and this seems a good way to offer silliness that isn't just killing someone.

## Changelog

:cl:
add: Being run over by the clown car will flatten you like a pancake. Honk!
soundadd: adds cartoon_splat.ogg
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
this is funny. Good feature

File: code/modules/vehicles/cars/clowncar.dm [hunk 1]
```diff
@@ -115,6 +115,16 @@
...
 	log_combat(src, bumped, "crashed into", null, "dumping all passengers")
 
+/obj/vehicle/sealed/car/clowncar/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if(has_gravity())
+		for(var/mob/living/carbon/human/target_pancake in loc)
+			if(target_pancake.body_position == LYING_DOWN && !HAS_TRAIT(target_pancake, TRAIT_INCAPACITATED))
+				target_pancake.visible_message(span_warning("[src] runs over [target_pancake], flattening [target_pancake.p_them()] like a pancake!"))
+				target_pancake.AddElement(/datum/element/squish, 10 SECONDS)
+				target_pancake.Paralyze(40)
```
Comment (Iamgoofball) [inline code/modules/vehicles/cars/clowncar.dm:125]: a 4 second hardstun for getting run over is extremely strong

Comment (TerraGS) [reply code/modules/vehicles/cars/clowncar.dm:125]: Was not sure what increment the parameter was so I just used the default climb stun. I can lower it if this is an issue.

**Comment by SkeletalElite** [comment]
Crawling is the counter to clown car. Letting clown car chain hardstun anyone who doesn't allow themselves to be stuck inside of it seems like a bad idea.

**Comment by JohnFulpWillard** [comment]
vry funy .. . 👍 

this helps clowns scoop people up from the floor, which is a good idea, lying down is lame.

**Comment by MrDoomBringer** [comment]
> Crawling is the counter to clown car. Letting clown car chain hardstun anyone who doesn't allow themselves to be stuck inside of it seems like a bad idea.

TBF, keeping the clown-car in one place only increases your chances of the AI bolting you in / getting your ride shot at

**Comment by moocowswag** [comment]
Its definitely a balance change but a welcome one.

**Comment by private-tristan** [comment]
how long is the stun time compared to the 'force person into car' timer, if clowns can hardstun you and immediately force you into car fighting clowncar would be quite difficult

**Comment by TerraGS** [comment]
> how long is the stun time compared to the 'force person into car' timer, if clowns can hardstun you and immediately force you into car fighting clowncar would be quite difficult

they are the same time, so you'd have to be very precise to scoop them while stunned

File: code/modules/vehicles/cars/clowncar.dm [hunk 2]
```diff
@@ -115,6 +115,16 @@
 	dump_mobs(TRUE)
 	log_combat(src, bumped, "crashed into", null, "dumping all passengers")
 
+/obj/vehicle/sealed/car/clowncar/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if(has_gravity())
+		for(var/mob/living/carbon/human/target_pancake in loc)
+			if(target_pancake.body_position == LYING_DOWN && !HAS_TRAIT(target_pancake, TRAIT_INCAPACITATED))
```
Comment (Zonespace27) [inline code/modules/vehicles/cars/clowncar.dm:122]: you could make this an early continue

File: code/modules/vehicles/cars/clowncar.dm [hunk 3]
```diff
@@ -115,6 +115,16 @@
...
+				target_pancake.visible_message(span_warning("[src] runs over [target_pancake], flattening [target_pancake.p_them()] like a pancake!"))
+				target_pancake.AddElement(/datum/element/squish, 5 SECONDS)
+				target_pancake.Paralyze(20)
```
Comment (Fikou) [inline code/modules/vehicles/cars/clowncar.dm:125]: use SECONDS defines

File: code/modules/vehicles/cars/clowncar.dm (hunk 2 above)
Comment (Fikou) [reply code/modules/vehicles/cars/clowncar.dm:122]: do that

File: code/modules/vehicles/cars/clowncar.dm [hunk 4]
```diff
@@ -115,6 +115,16 @@
 	dump_mobs(TRUE)
 	log_combat(src, bumped, "crashed into", null, "dumping all passengers")
 
+/obj/vehicle/sealed/car/clowncar/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if(has_gravity())
```
Comment (Fikou) [inline code/modules/vehicles/cars/clowncar.dm:120]: early return this

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/modules/vehicles/cars/clowncar.dm [hunk 5]
```diff
@@ -115,6 +115,20 @@
 	dump_mobs(TRUE)
 	log_combat(src, bumped, "crashed into", null, "dumping all passengers")
 
+/obj/vehicle/sealed/car/clowncar/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
```
Comment (JohnFulpWillard) [inline code/modules/vehicles/cars/clowncar.dm:118]: You can probably register this to ``COMSIG_MOVABLE_CROSS_OVER`` instead

File: code/modules/vehicles/cars/clowncar.dm [hunk 6]
```diff
@@ -127,6 +146,8 @@
 /obj/vehicle/sealed/car/clowncar/atom_destruction(damage_flag)
 	playsound(src, 'sound/vehicles/clowncar_fart.ogg', 100)
 	STOP_PROCESSING(SSobj,src)
+	UnregisterSignal(src, COMSIG_MOVABLE_BUMP)
+	UnregisterSignal(src, COMSIG_MOVABLE_CROSS_OVER)
```
Comment (JohnFulpWillard) [inline code/modules/vehicles/cars/clowncar.dm:150]: I'd prefer if these were on Destroy() instead

**Comment by at996** [comment]
I tried out the new clown car, apparently it just stopped having the ability to crash into walls and ram people into it, i really hope this isn't a intentional thing.

**Comment by theta060** [comment]
c*ders not testing their changes before they merge

**Comment by TerraGS** [comment]
I'm fixing it

**Comment by OdieLetegivre** [comment]
while the flattening is funny, a hardstun seems a biiiit ridiculous. should be changed to a brief softstun.

**Pull request merged**